hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c

```diff
@@ -2448,6 +2448,67 @@ int launch_container_as_user(const char *user, const char *app_id,
     return exit_code;
 }
 
+static int parse_int (const char *str, int *val)
+{
+  long num;
+  char *endptr;
+
+  errno = 0;
+  num = strtol(str, &endptr, 0);
+
+  if (errno)
+    return -errno;
+
+  if (endptr == str || *endptr != '\0')
+    return -EINVAL;
+
+  if (num < INT_MIN || num > INT_MAX)
+    return -EINVAL;
+
+  *val = num;
+  return 0;
+}
+
+static void kill_session (pid_t session, int sig)
```

Comment:
pid is a provided as an int, is pid_t just an alias to int ?

Reply:
pid_t is an alias for a signed int (typedef).